source/common/http/http1/balsa_parser.cc

```diff
@@ -268,6 +268,7 @@ void BalsaParser::MessageDone() {
   }
   status_ = convertResult(connection_->onMessageComplete());
   framer_.Reset();
+  on_message_begin_called_ = false;
```

Comment:
completely out of scope for this change, okay to consider later: Would it be right to reset other members also, e.g., headers_done_ and/or others? IIUC, resetting on_message_begin_called_ sort of suggests that we can reuse BalsaParser for a subsequent message (hm, or maybe just subsequent calls to execute()?).

Reply:
Yes, I think that makes perfect sense, I'll look into it.

Reply:
Thanks for catching this!  I filed #24557 for resetting `headers_done_`.  I couldn't identify other members that would need to be reset.  I'm fixing this, among other things, with test coverage for plenty of weird cases, at #24594.